openapi/openapi_client_test.go

```diff
@@ -70,6 +71,56 @@ func TestAppendOperationHeaders(t *testing.T) {
 	})
 }
 
+func TestAppendUserAgentHeader(t *testing.T) {
+	Convey("Given a providerClient and user agent header value", t, func() {
+		providerClient := &ProviderClient{}
+		expectedHeaderValue := "some user agent header value"
+		Convey("When appendUserAgentHeader with empty header map and header value", func() {
+			headers := map[string]string{}
+			providerClient.appendUserAgentHeader(headers, expectedHeaderValue)
+			Convey("Then the user agent header value should exist in the header map with correct value", func() {
+				value, exists := headers["User-Agent"]
+				So(exists, ShouldBeTrue)
+				So(value, ShouldEqual, expectedHeaderValue)
+			})
+		})
+		Convey("When appendUserAgentHeader with non-empty header map and header value", func() {
+			headers := map[string]string{"Some-Header": "some header value"}
+			providerClient.appendUserAgentHeader(headers, expectedHeaderValue)
+			Convey("Then the user agent header should exist in the header map with correct value", func() {
+				value, exists := headers["User-Agent"]
```

Comment:
should we also confirm here that the already existing item "Some-Header also exists in the map? to make sure the function for some reason is not removing elements etc